packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart

```diff
@@ -48,12 +50,23 @@ class AppSizeController {
   /// The node set as the analysis tab root.
   ///
   /// Used to build the treemap and the tree table for the analysis tab.
-  ValueListenable<TreemapNode?> get analysisRoot => _analysisRoot;
-  final _analysisRoot = ValueNotifier<TreemapNode?>(null);
+  final analysisRoot =
+      ValueNotifier<Selection<TreemapNode>>(Selection<TreemapNode>());
 
   void changeAnalysisRoot(TreemapNode? newRoot) {
-    _analysisRoot.value = newRoot;
-    if (newRoot == null) return;
+    if (newRoot == null) {
+      analysisRoot.value = Selection<TreemapNode>();
+      return;
+    }
+
+    final searchCondition = (TreemapNode n) => n == newRoot;
+    final int selectedRowIndex = newRoot.root.childCountToMatchingNode(
```

Comment:
I'm pretty sure `newRoot.root` can be changed to just `newRoot` (`newRoot` should be the root node and therefore shouldn't have any `parent` nodes)

Reply:
I believe newRoot is a node that is being passed as the selected node and not the root node (it's always changing).

Reply:
Ah got it - let's maybe rename to `newAnalysisRoot` for clarity